File: lib/src/test/java/net/mm2d/upnp/empty/EmptyServiceTest.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty;

import net.mm2d.upnp.Service;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import java.io.IOException;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.*;

@SuppressWarnings("NonAsciiCharacters")
@RunWith(JUnit4.class)
public class EmptyServiceTest {
    @Test
    public void getDevice() {
        final Service service = new EmptyService();
        assertThat(service.getDevice(), is(not(nullValue())));
    }

    @Test
    public void getServiceType() {
        final Service service = new EmptyService();
        assertThat(service.getServiceType(), is(not(nullValue())));
    }

    @Test
    public void getServiceId() {
        final Service service = new EmptyService();
        assertThat(service.getServiceId(), is(not(nullValue())));
    }

    @Test
    public void getScpdUrl() {
        final Service service = new EmptyService();
        assertThat(service.getScpdUrl(), is(not(nullValue())));
    }

    @Test
    public void getControlUrl() {
        final Service service = new EmptyService();
        assertThat(service.getControlUrl(), is(not(nullValue())));
    }

    @Test
    public void getEventSubUrl() {
        final Service service = new EmptyService();
        assertThat(service.getEventSubUrl(), is(not(nullValue())));
    }

    @Test
    public void getDescription() {
        final Service service = new EmptyService();
        assertThat(service.getDescription(), is(not(nullValue())));
    }

    @Test
    public void getActionList() {
        final Service service = new EmptyService();
        assertThat(service.getActionList(), is(not(nullValue())));
    }

    @Test
    public void findAction() {
        final Service service = new EmptyService();
        assertThat(service.findAction(""), is(nullValue()));
    }

    @Test
    public void getStateVariableList() {
        final Service service = new EmptyService();
        assertThat(service.getStateVariableList(), is(not(nullValue())));
    }

    @Test
    public void findStateVariable() {
        final Service service = new EmptyService();
        assertThat(service.findStateVariable(""), is(nullValue()));
    }

    @Test(expected = IOException.class)
    public void subscribe() throws Exception {
        final Service service = new EmptyService();
        service.subscribe();
    }

    @Test(expected = IOException.class)
    public void subscribe1() throws Exception {
        final Service service = new EmptyService();
        service.subscribe(true);
    }

    @Test(expected = IOException.class)
    public void renewSubscribe() throws Exception {
        final Service service = new EmptyService();
        service.renewSubscribe();
    }

    @Test(expected = IOException.class)
    public void unsubscribe() throws Exception {
        final Service service = new EmptyService();
        service.unsubscribe();
    }

    @Test
    public void expired() {
        final Service service = new EmptyService();
        service.expired();
    }

    @Test
    public void getSubscriptionId() {
        final Service service = new EmptyService();
        assertThat(service.getSubscriptionId(), is(nullValue()));
    }

    @Test
    public void getSubscriptionStart() {
        final Service service = new EmptyService();
        assertThat(service.getSubscriptionStart(), is(0L));
    }

    @Test
    public void getSubscriptionTimeout() {
        final Service service = new EmptyService();
        assertThat(service.getSubscriptionTimeout(), is(0L));
    }

    @Test
    public void getSubscriptionExpiryTime() {
        final Service service = new EmptyService();
        assertThat(service.getSubscriptionExpiryTime(), is(0L));
    }
}

File: mmupnp/src/test/java/net/mm2d/upnp/cp/empty/EmptyServiceTest.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.cp.empty

import com.google.common.truth.Truth.assertThat
import io.mockk.every
import io.mockk.mockk
import io.mockk.slot
import kotlinx.coroutines.runBlocking
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class EmptyServiceTest {
    @Test
    fun getDevice() {
        val service = EmptyService
        assertThat(service.device).isNotNull()
    }

    @Test
    fun getServiceType() {
        val service = EmptyService
        assertThat(service.serviceType).isNotNull()
    }

    @Test
    fun getServiceId() {
        val service = EmptyService
        assertThat(service.serviceId).isNotNull()
    }

    @Test
    fun getScpdUrl() {
        val service = EmptyService
        assertThat(service.scpdUrl).isNotNull()
    }

    @Test
    fun getControlUrl() {
        val service = EmptyService
        assertThat(service.controlUrl).isNotNull()
    }

    @Test
    fun getEventSubUrl() {
        val service = EmptyService
        assertThat(service.eventSubUrl).isNotNull()
    }

    @Test
    fun getDescription() {
        val service = EmptyService
        assertThat(service.description).isNotNull()
    }

    @Test
    fun getActionList() {
        val service = EmptyService
        assertThat(service.actionList).isNotNull()
    }

    @Test
    fun findAction() {
        val service = EmptyService
        assertThat(service.findAction("")).isNull()
    }

    @Test
    fun getStateVariableList() {
        val service = EmptyService
        assertThat(service.stateVariableList).isNotNull()
    }

    @Test
    fun findStateVariable() {
        val service = EmptyService
        assertThat(service.findStateVariable("")).isNull()
    }

    @Test
    fun getSubscriptionId() {
        val service = EmptyService
        assertThat(service.subscriptionId).isNull()
    }

    @Test
    fun subscribeSync() {
        val service = EmptyService
        assertThat(service.subscribeSync()).isFalse()
    }

    @Test
    fun subscribeSync1() {
        val service = EmptyService
        assertThat(service.subscribeSync(true)).isFalse()
    }

    @Test
    fun renewSubscribeSync() {
        val service = EmptyService
        assertThat(service.renewSubscribeSync()).isFalse()
    }

    @Test
    fun unsubscribeSync() {
        val service = EmptyService
        assertThat(service.unsubscribeSync()).isFalse()
    }

    @Test
    fun subscribe() {
        val service = EmptyService
        val callback: (Boolean) -> Unit = mockk()
        val slot = slot<Boolean>()
        every { callback.invoke(capture(slot)) } answers { nothing }
        service.subscribe(callback = callback)
        assertThat(slot.captured).isFalse()
    }

    @Test
    fun subscribe1() {
        val service = EmptyService
        val callback: (Boolean) -> Unit = mockk()
        val slot = slot<Boolean>()
        every { callback.invoke(capture(slot)) } answers { nothing }
        service.subscribe(true, callback)
        assertThat(slot.captured).isFalse()
    }

    @Test
    fun subscribe2() {
        val service = EmptyService
        service.subscribe(callback = null)
    }

    @Test
    fun subscribe3() {
        val service = EmptyService
        service.subscribe(true, null)
    }

    @Test
    fun renewSubscribe() {
        val service = EmptyService
        val callback: (Boolean) -> Unit = mockk()
        val slot = slot<Boolean>()
        every { callback.invoke(capture(slot)) } answers { nothing }
        service.renewSubscribe(callback)
        assertThat(slot.captured).isFalse()
    }

    @Test
    fun renewSubscribe1() {
        val service = EmptyService
        service.renewSubscribe(null)
    }

    @Test
    fun unsubscribe() {
        val service = EmptyService
        val callback: (Boolean) -> Unit = mockk()
        val slot = slot<Boolean>()
        every { callback.invoke(capture(slot)) } answers { nothing }
        service.unsubscribe(callback)
        assertThat(slot.captured).isFalse()
    }

    @Test
    fun unsubscribe1() {
        val service = EmptyService
        service.unsubscribe(null)
    }

    @Test
    fun subscribeAsync() {
        val service = EmptyService
        runBlocking {
            assertThat(service.subscribeAsync()).isFalse()
        }
    }

    @Test
    fun subscribeAsync1() {
        val service = EmptyService
        runBlocking {
            assertThat(service.subscribeAsync(true)).isFalse()
        }
    }

    @Test
    fun renewSubscribeAsync() {
        val service = EmptyService
        runBlocking {
            assertThat(service.renewSubscribeAsync()).isFalse()
        }
    }

    @Test
    fun unsubscribeAsync() {
        val service = EmptyService
        runBlocking {
            assertThat(service.unsubscribeAsync()).isFalse()
        }
    }
}
